python/jupytergis_lab/src/notebookrenderer.ts

```diff
@@ -86,20 +153,50 @@ export const notebookRendererPlugin: JupyterFrontEndPlugin<void> = {
           sharedModel: sharedModel as IJupyterGISDoc
         });
 
+        const currentWidget = app.shell.currentWidget;
+        if (
+          currentWidget instanceof NotebookPanel ||
+          currentWidget instanceof ConsolePanel
+        ) {
+          this.parent = currentWidget;
+        }
+
+        this.parent?.disposed.connect(this.dispose);
+        this.jupyterGISModel.contentsManager = app.serviceManager.contents;
+
+        if (!sharedModel) {
+          // The path of the project is set to the path of the notebook, to be able to
+          // add local geoJSON/shape file in a "file-less" project.
+          let currentWidgetPath: string | undefined = undefined;
+          currentWidgetPath = this.parent?.sessionContext.path;
+          if (currentWidgetPath) {
+            this.jupyterGISModel.filePath = PathExt.join(
+              PathExt.dirname(currentWidgetPath),
+              'unsaved_project'
```

Comment:
Instead of giving this file path, what would it take to create the file from the front-end? If no file name is provided, we create an "untitled" file, if a new file path is provided (something we don't know) we create the file.

In replacement of what is removed in https://github.com/geojupyter/jupytergis/pull/426/files

Reply:
I think it is also nice to be able to debug without creating a file. And to add a `save()` method to the API, to be able to save the document.
But indeed, if the path is provided and does not exist, we could create it from the frontend.